File: leetcode/cpp/884_Uncommon_Words_from_Two_Sentences.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <sstream>
using namespace std;
class Solution {
public:
	vector<string> uncommonFromSentences(string s1, string s2) {
		unordered_map<string, int> mp;
		stringstream ss1(s1), ss2(s2);
		string word;

		while (ss1 >> word) mp[word]++;
		while (ss2 >> word) mp[word]++;
		
		vector<string> ret;
		for (auto [w, cnt] : mp) {
			// appears more than once
			if (cnt > 1) continue;
			ret.push_back(w);
		}

		return ret;
	}
};

File: leetcode/cpp/884_Uncommon_Words_from_Two_Sentences.py
from collections import Counter
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        mp = Counter()

        for word in s1.split():
            mp[word] += 1

        for word in s2.split():
            mp[word] += 1

        ret = []
        for word, cnt in mp.items():
            # appears only once across both sentences
            if cnt == 1:
                ret.append(word)

        return ret
